tools/api-performance-test/performance.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python3
+"""Mock API queries and send them to the test API endpoint for
+performance testing. It is recommended to use two terminals to
+run this script concurrently to generate sufficient traffic."""
+
+import aiohttp
+import asyncio
+import os
+import random
+import sys
+import time
+import json
+
+from google.cloud import ndb
+import osv
+
+from collections import Counter, defaultdict
+
+BASE_URL = 'https://api.test.osv.dev/v1'
+TOTAL_RUNTIME = 3600  # total run time in second
+GCP_PROJECT = 'oss-vdb-test'
+BUG_DIR = './all_bugs'
+
+# Number of `vulnerability get` requests to send per second
+VULN_QUERY_BATCH_SIZE = 50
+# Number of `version query` requests to send per second
+VERSION_QUERY_BATCH_SIZE = 100
+# Number of `package query` requests to send per second
+PACKAGE_QUERY_BATCH_SIZE = 30
+# Number of `batch query` requests to send per second
+BATCH_QUERY_BATCH_SIZE = 3
+# Number of large `batch query` requests to send per second
+LARGE_BATCH_QUERY_BATCH_SIZE = 2
+
+
+class SimpleBug:
+  """A simplified bug only contains essential information
+  for making HTTP requests."""
+
+  def __init__(self, bug_dict):
+    self.db_id = bug_dict['db_id']
+    # If the package/ecosystem/version value is None, then add a fake value in.
+    if not bug_dict['project']:
+      self.packages = 'foo'
+    else:
+      self.packages = list(bug_dict['project'])
+    self.purl = bug_dict['purl']
+    if not bug_dict['ecosystem']:
+      self.ecosystems = 'foo'
+    else:
+      self.ecosystems = list(bug_dict['ecosystem'])
+
+    # Use the `affected fuzzy` value as the query version.
+    # If no 'affected fuzzy' is present, assign a default value.
+    self.affected_fuzzy = bug_dict['affected_fuzzy']
+    if not self.affected_fuzzy:
+      self.affected_fuzzy = '1.0.0'
+
+
+def format_bug_for_output(bug) -> dict[str:any]:
```

Comment:
pretty sure this should be `dict[str, any]`

Reply:
Updated